proposals/ref-struct-interfaces.md

```diff
@@ -147,61 +151,113 @@ The new rules apply to the `heap` constraint:
 
-While that is true it can present a problem in multi-targeted scenarios. Code would compile in one target framework but fail in another. This could lead to confusion with customers and result in a desire for a more explicit opt-in.
+This auto application of `allows ref struct` poses a few problems though. The first is in multi-targeted scenarios. Code would compile in one target framework but fail in another and there is no syntactic indicator of why the APIs should behave differently.
+
+```csharp
+// Works in net9.0 but fails in all other TF
+Func<Span<char>> func;
+```
+
+This is likely to lead to customer confusion and looking at changes in `Func<T>` in the `net9.0` source wouldn't give customers any clue as to what changed.
+
+The other issue is that very subtle changes in code can cause _spooky action at a distance_ problems. Consider the following code:
+
+```csharp
+interface I1<T>
+{
+}
+```
+
+This interface would be eligable for auto-application of `allows ref struct`. If a developer comes around later though and adds a default interface method then suddenly it would not be and it would break any consumers that had already created invocations like `I1<Span<char>>`. This is a very subtle change that would be hard to track down.
 
 ### Binary breaking change
-Adding `allow T: ref struct` to an existing API is not a source breaking change. It is purely expanding the set of allowed types for an API. Need to track down if this is a binary breaking change or not. Unclear if updating the attributes of a generic parameter constitute a binary breaking change.
+
+Adding `allows ref struct` to an existing API is not a source breaking change. It is purely expanding the set of allowed types for an API. Need to track down if this is a binary breaking change or not. Unclear if updating the attributes of a generic parameter constitute a binary breaking change.
 
 ## Considerations
 
 ### Runtime support
+
 This feature requires several pieces of support from the runtime / libraries team:
+
 - Preventing default interface methods from applying to `ref struct`
-- API in `System.Reflection.Metadata` for encoding the `gpAcceptByRefLike` value.
+- API in `System.Reflection.Metadata` for encoding the `gpAcceptByRefLike` value
 - Support for generic parameters being a `ref struct`
 
-Most of this support is likely already in place. The general `ref struct` as generic parameter support is already implemented as described [here](https://github.com/dotnet/runtime/blob/main/docs/design/features/byreflike-generics.md). It's possible the DIM implementation already account for `ref struct`. But each of these items needs to be tracked down.
+Most of this support is likely already in place. The general `ref struct` as generic parameter support is already implemented as described [here][byref-like-generics]. It's possible the DIM implementation already account for `ref struct`. But each of these items needs to be tracked down.
+
+## API versioning
```

Comment:
@stephentoub section I promised about how I think the anti-constraint impacts API versioning.